src/libAtomVM/context.c

```diff
@@ -123,10 +123,31 @@ void context_destroy(Context *ctx)
 
     // When monitor message is sent, process is no longer in the table
     // and is no longer registered either.
-    context_monitors_handle_terminate(ctx);
+    struct ResourceMonitor *resource_monitor = context_monitors_handle_terminate(ctx);
 
     synclist_unlock(&ctx->global->processes_table);
 
+    // Eventually call resource monitors handlers after the processes table was unlocked
+    // The monitors were removed from the list of monitors.
+    if (resource_monitor) {
+        ErlNifEnv env;
+        erl_nif_env_partial_init_from_globalcontext(&env, ctx->global);
+
+        struct ListHead monitors;
+        list_prepend(&resource_monitor->base.monitor_list_head, &monitors);
+
+        struct ListHead *item;
+        struct ListHead *tmp;
+        MUTABLE_LIST_FOR_EACH (item, tmp, &monitors) {
+            struct Monitor *monitor = GET_LIST_ENTRY(item, struct Monitor, monitor_list_head);
+            resource_monitor = (struct ResourceMonitor *) monitor;
```

Comment:
This is not safe in case struct layout is changed, let's rather use `CONTAINER_OF`.